test/tests/conditional_tests.py

```diff
@@ -0,0 +1,166 @@
+import io
+import os
+import pavilion
+
+from pavilion import arguments
+from pavilion import commands
+from pavilion import plugins
+from pavilion.status_file import STATES
+from pavilion.test_config import file_format, setup, variables
+from pavilion.test_run import TestRun, TestRunError
+from pavilion import unittest
+
+
+class conditionalTest(unittest.PavTestCase):
+
+    def setUp(self):
+        plugins.initialize_plugins(self.pav_cfg)
+
+    def tearDown(self):
+        plugins._reset_plugins()
+
+    def test_success(self):  # this method runs some conditional successes
+        # using the variables to test logic of not_if and only if
+        test_list = []
+        test_cfg = self._quick_test_cfg()
+        test_cfg['run']['cmds'] = ['echo "Goodbye World"']
+
+        test_cfg = {'variables': {'person': ['calvin'],
+                                  'machine': ['bieber']},
+                    'not_if': {'person': ['bleh', 'notcalvin']},
```

Comment:
A proper logical check of this should include two keys for each of the `not_if` and `only_if`. Otherwise, you're not completely checking the logic in this case.